### ISSUE TITLE
Feature request- upload photos in folder hierarchy from date

### ISSUE THREAD

**Issue by lucasvieites** [owncloud/android#2197]
### Actual behaviour
- Uploading photos or videos from a phone stores everything in one folder

### Desired behaviour
- Be able to choose uploading into a file hierarchy derived from the photo's date, for example:
-2018
|--01
|--02
|--03
[...]
|--12
-2017
|--01
|--02
[...]
|--12
 ...etc
 
### Steps to reproduce
Not applicable

**Comment by davigonz** [comment]
Hi @lucasvieites , thank you for your suggestion but I don't think that implementing this option is one of our priorities at this moment. Anyway, contributions are always welcome 😉

**Comment by lucasvieites** [comment]
Thanks for your reply, mobile development is not in my skillset but I'll look into creating a plugin around this idea.

**Comment by sd1998** [comment]
@davigonz I would like to work on this

**Comment by sd1998** [comment]
@jesmrec Should I add this as an option in the settings and if the user enables this option only then the images will be saved based on file hierarchy derived from the image date?

**Comment by jesmrec** [comment]
The idea of this feature sounds interesting for the case the app is installed from scratch or the instant uploads are enabled/triggered for the first time. All pics are organized using folders by timestamp year-month. 

But... If i have enabled it before, i will have my folder with all my pics together (current behaviour). From the moment this feature is enabled, my pics will be stored in different folders. I will have pics with no folder and pics with folder. IMHO this is kind of mess. 

Any idea about?

**Comment by lucasvieites** [comment]
Hi @sd1998 , @jesmrec,
thank you for working on this!
I think we are talking about two different things;

1. New install - Enable "timestamp hierarchy"
2. Existing installation - Re-order existing uploads in "timestamp hierarchy"

Option 1 is clear, and for option 2 I'd suggest having a greyed out setting that get's writable when you enable "timestamp hierarchy" and asks if the user wants to re-order the existing uploads. Like in the "[Dependency](http://www.androiddocs.com/design/patterns/settings.html)" setting type.

Where are you planning to take the timestamp from? The file's creation date is not always reliable. Perhaps from EXIF information (if it exists) or the filename, could be a better source.

Thank you and kind regards,
Lucas

**Comment by theScrabi** [comment]
> Existing installation - Re-order existing uploads in "timestamp hierarchy"

Hm well reordering in that great size should be made part of the server what do you thing?

**Comment by davigonz** [comment]
If I have understood correctly, what @lucasvieites suggests is the next scenario:

1. Press upload button.
2. Select a couple of pictures, e.g. **IMG_20180721, IMG_20180615, IMG_20170322**
3. App automatically reads the file creation or last modification date of those pictures, create the next hierarchy in internal storage (ownCloud folder) and upload the files following the same hierarchy on the server:
- 2017
  - March
    - **IMG_20170322**
- 2018
  - June
    - **IMG_20180615**
  - July
    - **IMG_20180721**

Is this correct?

**Comment by lucasvieites** [comment]
@theScrabi You're absolutely right! What was I thinking?! That should not be done from the mobile side.
@davigonz Yes, that's the general idea. I'm not sure if the pictures are also stored locally on the device.

Thank you

**Comment by butonic** [comment]
I would love to collect all media files and sync them to a timeline tree (`/<Devicename> Media/<YYYY>/<MM>/IMG_YYYYMMDD_HHMMSS.ext` by default). Unfortunately, pctures on the device might be stored in different folders, eg. the camera app and whatsapp store images in separate folders under separate apps. 

I see two aspects that we need to consider:
1. We should listen on the media scanner for any new files to get allow syncing all media that is created. Being able to only one folder sucks. Having to configure all possible locations is not grandparent compatible (as proposed in https://github.com/owncloud/android/issues/2396). 
2. What about existing files? Leave them untouched. I already moved my pictures to different folders because 7 years of images in a single folder makes the PROPFIND on that folder unusable and cause timeouts. If users really want it we can add a media manager app that can be used to manage photos. That would manage a timeline tree ... not the responsibility of the android app.

**Comment by jesmrec** [comment]
For the moment, this would be applied only for the camera uploads.

**Comment by DeepDiver1975** [comment]
🤖 Automated issue triage by Claude Code review agent.

This issue was opened before 2020 and has had no activity since then, so we are
closing it as outdated. If this is still relevant on a current ownCloud release,
please **reopen** (or open a fresh issue) with up-to-date details — thank you\!
